### ISSUE TITLE
How to set DATESTYLE on new connections

### ISSUE THREAD

**Issue by googol** [porsager/postgres#645]
Hi,

I'm looking for a way to automatically set DATESTYLE on all new connections opened by the node pool, to facilitate parsing timestamp values into temporal instants, but I'm not finding anything that could do that.

node-postgres has the [`connect`](https://node-postgres.com/apis/pool#connect) event on it's node pool that allows executing sql on all new connections before they are used, is there anything similar, or would you be open to adding that sort of functionality?

**Comment by googol** [comment]
Oh that was simpler than expected, thanks!

I had a hunch that the connection prop could be related, it's just that only the application_name is documented anywhere. Is there some list of what connection parameters can be set via this object?

**Comment by porsager** [comment]
Hehe, yeah 😊 

There's quite a good list here : https://www.postgresql.org/docs/current/runtime-config-client.html

Would be good to have a section in the docs for this, you're mouthe first to ask. What did you search for in the readme before posting here?

**Comment by googol** [comment]
I looked for the specific thing, datestyle, then `set search_path` with a couple of variations, then scanned for anything about connection startup. Looked for those in the issues as well.

Adding a link to that documentation page in the documentation about the `connection` config prop would be a good start with regard to documenting it's use. Is application_name something specific to postgresjs btw, it's not listed on that page? 

In the end my issue wasn't about the datestyle after all, it was just that I had to override the default type parser for timestamps in postgresjs. The project I'm working on uses Drizzle with postgresjs, and it has it's own layer of type conversions on top. I was using that to convert my timestamps to Temporal.Instants. The documentation and source indicated that the builtin timestamp parser converts from string to Date, but the incoming value had actually been converted to a Date already by postgresjs. Maybe this helps someone with similar issues :D
